costmodel/promparsers.go

```diff
@@ -148,6 +148,11 @@ func parseDataPoint(dataPoint interface{}) (*Vector, error) {
 		return nil, err
 	}
 
+	if math.IsNaN(v) {
+		klog.V(1).Infof("Warning: Found NaN value parsing vector data point.")
```

Comment:
Any chance we could see labels here? I know you'd need to pass them in from somewhere but it would be useful to know what's NaN. otherwise there will be thousands of these statements with no real context.

Reply:
Yeah, we can do that...